docs: Improve the video playback guide (#246)
livepeer/livepeer-studio-docs: docs/04-guides/playback-a-video-stream.mdx

import CodeBlock from "@theme/CodeBlock";
import CodeBlockTabs from "@site/src/components/CodeBlockTabs";
import CodeBlockTabItem from "@site/src/components/CodeBlockTabItem";
import Tabs from "@theme/Tabs";
import TabItem from "@theme/TabItem";
import Player from "@site/src/components/Player/Livepeer";
import HTMLPlayer from "@site/src/components/Player/HTML";

# Playback a video

This guide provides you the instructions to playback a video or a live stream on
your website/applications. To playback live streams or on-demand video inside
your application, you'll need to use a video player component that supports the
HLS protocol.

You can either use Livepeer's built-in player or your own player. This guide
covers both options.

> ## Using Livepeer's player

Reviewer: Should we use a tabbed display, like the tutorials for API / dashboard? One for livepeer.js and one for "custom player".

Then we should later add one bringing back the "embeddable player" (lvpr.tv) docs as well, which is still the easiest way to integrate if you're not using React. Don't worry about this one now tho, we had docs for this in the old repo but they got removed on this `studio-docs` one. IMO we should bring them back, but it isn't a regression of this PR anyway.

Livepeer.js is a JavaScript library that allows you to upload, retrieve, and
play assets. It provides a core vanilla JS library to easily connect to a
Livepeer provider (e.g. Livepeer Studio), as well as React-specific
hooks/components to provide memoization and DOM management.

:::note

Livepeer.js currently only supports applications built with React.js and
Next.js.

> :::

Reviewer: We should probably link to livepeer.js home page here as something like "for more information visit livepeerjs.org" or something like that. Could be just making the `Livepeer.js` hyperlinks to that page as well.

Or perhaps maybe in the end of this livepeer.js player section we should link to this guide: https://livepeerjs.org/react/Player

Then one could get more info on everything they could configure on the player etc. Could be right after the simple player example.

Reviewer: FTR here was the old embeddable player docs
https://github.com/livepeer/livepeer-com/blob/36495e35a4ffa945cd206754139118136ca31fa5/packages/www/docs/guides/start-live-streaming/playback.mdx#L31

  </TabItem>
  <TabItem value="yarn" label="yarn">
    <CodeBlock language="bash">yarn add @livepeer/react</CodeBlock>
  </TabItem>
</Tabs>

### Usage

The Player component provides an easy way to display video or audio. You can
also customize the player's theme and controls.

```jsx
import { Player } from "@livepeer/react";

export const SimplePlayer = () => {
  return (
    <Player
      title="Agent 327: Operation Barbershop"
      playbackId="6d7el73r1y12chxr"
      showPipButton
    />
  );
};
```

<Player />

## Using your own player

Using livepeer.js is the easiest and recommended way to playback a video or a
live stream. However, if you want to use your own player, you can do so by
following the instructions below.

:::caution

Please note that to playback live streams inside your application you'll need to
use a video player component that supports the HLS.

:::

### Create a playback URL

To playback a live stream in other players, you'll need to create an HLS URL.
HLS is a protocol that allows you to stream video and audio content over HTTP.
Most of the video you watch on the web is delivered using HLS. Livepeer also
uses HLS to deliver video and audio content.

> You can create a HLS url by using the below format and replacing the

Reviewer: ```suggestion
You can create an HLS url by using the below format and replacing the
```

> You can create a HLS url by using the below format and replacing the

Reviewer: I think before giving the example, it's worth making a short remark that the playback URLs are different for streams and on-demand assets. Just so one doesn't grab the first example and start using it only to realize later it doesn't work.

`PLAYBACK_ID` with your stream's playback id.

```txt
https://livepeercdn.com/hls/{PLAYBACK_ID}/index.m3u8
```

> However, if you want to playback a video that is already uploaded to Livepeer,
> you can use the below format and replace the `PLAYBACK_ID` with your video's
> playback id.

Reviewer: I think it would be clearer like this, wdyt?:
```suggestion
On-demand videos have a different format for the playback URLs. To playback such videos,
you can use the below format and replace the `PLAYBACK_ID` with your video's
playback id.
```
(this should be different if you make the remark about the different playback URLs before giving any exampls)

Author: sounds great!

```txt
https://livepeercdn.com/asset/{PLAYBACK_ID}/video
```

### Use the playback URL in a player

You can use the playback URL is any video player that supports HLS. Here is a
list of popular players that supports HLS:

- [Video.js](https://videojs.com/)
- [Plyr.io](https://plyr.io/)
- [HLS.js](https://github.com/video-dev/hls.js)
- [JW Player](https://www.jwplayer.com/html5-video-player/)
- [Bitmovin Player](https://bitmovin.com/video-player/)

Here is an example of how to use the playback URL in a `video` tag.

```jsx
import React from "react";

export default function Player() {
  return (
    <div>
      <video
        controls

>         src="https://livepeercdn.com/asset/6d7el73r1y12chxr/video"

Reviewer: Hmm does this work? I think it will only work on Apple devices by default. We might want to give a working example here, maybe with video.js or hls.js.
